src/components/ChartButton.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+interface IProps {
+  locations: string[];
+  handleChangeLocation: (value: string) => void;
+  currentParams: string | null;
+}
+
+export default function ChartButton({ locations, handleChangeLocation, currentParams }: IProps) {
+  return (
+    <div className="mb-6">
+      {[...locations, '필터해제']?.map((item, i) => {
+        return currentParams === item ? (
+          <button
+            className="bg-[#38A5FF] text-base text-white font-bold py-1 px-3 rounded-full mx-1"
+            key={i}
+            onClick={() => handleChangeLocation(item)}
+          >
+            {item}
+          </button>
+        ) : (
+          <button
+            className="bg-[#87D5FF] hover:bg-[#38A5FF] text-base text-white font-bold py-1 px-3 rounded-full mx-1"
+            key={i}
+            onClick={() => handleChangeLocation(item)}
+          >
+            {item}
+          </button>
+        );
+      })}
```

Comment:
중복되는 부분이 많아보여요~
실제로 달라지는 className부분만 분리해서 전달하는 방식은 어떨까요??
```suggestion
       {[...locations, '필터해제'].map((item, i) => {
        const className =
          currentParams === item
            ? 'bg-[#38A5FF] text-base text-white font-bold py-1 px-3 rounded-full mx-1'
            : 'bg-[#87D5FF] hover:bg-[#38A5FF] text-base text-white font-bold py-1 px-3 rounded-full mx-1';

        return (
          <button className={className} key={i} onClick={() => handleChangeLocation(item)}>
            {item}
          </button>
        );
      })}
```


그리고 [...locations, '필터해제'] 배열은 항상 존재하니까 ?는 없어도 괜찮을거같아요!